src/attr/_make.py

```diff
@@ -450,6 +450,9 @@ def _transform_attrs(
     if field_transformer is not None:
         attrs = field_transformer(cls, attrs)
 
+        if inspect.isgenerator(attrs):
+            attrs = tuple(attrs)
```

Comment:
```suggestion
        if not isinstance(attrs, (list, tuple)):
            attrs = tuple(attrs)
```

Reply:
It does pass without the isinstance check, but it seems more wasteful to turn a list into a tuple when? Dunno

Reply:
I've tested both the suggestions locally, they both work for everything covered in the tests, so pick which ever you like best I think @hynek